syftlib/src/test/java/org/openmined/syft/unit/monitor/network/NetworkStatusRealDataSourceTest.kt

```diff
@@ -0,0 +1,155 @@
+package org.openmined.syft.unit.monitor.network
+
+import android.accounts.NetworkErrorException
+import android.net.NetworkCapabilities
+import org.junit.Test
+import org.openmined.syft.common.AbstractSyftWorkerTest
+import org.openmined.syft.common.HttpClientMock
+import org.openmined.syft.common.SocketClientMock
+import org.openmined.syft.domain.SyftConfiguration
+import org.openmined.syft.monitor.network.NetworkStatusModel
+import org.openmined.syft.monitor.network.NetworkStatusRealTimeDataSource
+
+@ExperimentalUnsignedTypes
+class NetworkStatusRealDataSourceTest : AbstractSyftWorkerTest() {
+
+    private val workerId = "test id"
+    private val networkStatusModel = NetworkStatusModel()
+
+    private var networkDataSourceSuccess: NetworkStatusRealTimeDataSource
+    private var networkDataSourceFailure: NetworkStatusRealTimeDataSource
+
+    init {
+        val socketClient = SocketClientMock(
+            authenticateSuccess = true,
+            cycleSuccess = true
+        )
+
+        // Configure client to return success response
+        val httpClientMockSuccess = HttpClientMock(
+            pingSuccess = true, downloadSpeedSuccess = true,
+            uploadSuccess = true, downloadPlanSuccess = true, downloadModelSuccess = true
+        )
+
+        val configurationSuccess = SyftConfiguration(
+            context,
+            networkingSchedulers,
+            computeSchedulers,
+            context.filesDir,
+            true,
+            networkConstraints,
+            NetworkCapabilities.TRANSPORT_WIFI,
+            0,
+            1,
+            socketClient.getMockedClient(),
+            httpClientMockSuccess.getMockedClient(),
+            SyftConfiguration.NetworkingClients.HTTP
+        )
+
+        networkDataSourceSuccess = NetworkStatusRealTimeDataSource.initialize(configurationSuccess)
+
+
+        // Configure client to return failure response
+        val httpClientMockFailure = HttpClientMock(
+            pingSuccess = false, downloadSpeedSuccess = false,
+            uploadSuccess = false, downloadPlanSuccess = false, downloadModelSuccess = false
+        )
+
+        val configurationFailure = SyftConfiguration(
+            context,
+            networkingSchedulers,
+            computeSchedulers,
+            context.filesDir,
+            true,
+            networkConstraints,
+            NetworkCapabilities.TRANSPORT_WIFI,
+            0,
+            1,
+            socketClient.getMockedClient(),
+            httpClientMockFailure.getMockedClient(),
+            SyftConfiguration.NetworkingClients.HTTP
+        )
+
+        networkDataSourceFailure = NetworkStatusRealTimeDataSource.initialize(configurationFailure)
+    }
+
+    @Test
+    fun `updatePing return success response from server`() {
+        networkDataSourceSuccess
+                .updatePing(workerId, networkStatusModel)
```

Comment:
Instead of asserting for completion, It would be better to see if a non null ping value is returned. You may require to ask the thread to wait for sometime here